### ISSUE TITLE
Embedding entire dashboard in UI app issue

### ISSUE THREAD

**Issue by donnalmh** [atoti/atoti#338]
## Steps to reproduce

Implemented an iframe with the src as the link to the Atoti Dashboard
## Actual Result

iframe not able to render anything and request shows forbidden 403 

Heres the network call:
Request URL: http://127.0.0.1:8888/api/security/csp-report
Request Method: POST
Status Code: 403 Forbidden
Remote Address: 127.0.0.1:8888
Referrer Policy: strict-origin-when-cross-origin
Content-Length: 40
Content-Security-Policy: frame-ancestors 'self'; report-uri /api/security/csp-report; default-src 'none'
Content-Type: application/json
Date: Sun, 15 Aug 2021 16:12:21 GMT
Server: TornadoServer/6.1
X-Content-Type-Options: nosniff
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-GB,en-US;q=0.9,en;q=0.8
Connection: keep-alive
Content-Length: 360
Content-Type: application/csp-report
Host: 127.0.0.1:8888
Origin: http://localhost:3000
Referer: http://localhost:3000/
sec-ch-ua: "Chromium";v="92", " Not A;Brand";v="99", "Google Chrome";v="92"
sec-ch-ua-mobile: ?0
Sec-Fetch-Dest: report
Sec-Fetch-Mode: no-cors
Sec-Fetch-Site: cross-site
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36
{csp-report: {document-uri: "http://127.0.0.1:8888/", referrer: "http://localhost:3000/dashboard",…}}
csp-report: {document-uri: "http://127.0.0.1:8888/", referrer: "http://localhost:3000/dashboard",…}

## Expected Result
iframe should show the actual interactive atoti dashboard

**Comment by tibdex** [comment]
Hello.

Depending on how they are used, iframes could lead to security issues. The atoti web app is also not designed to live in an iframe. For instance, you won't be able to bookmark the URL of a saved dashboard since the URL visible by your browser will not be the one controlled by the app. There are often better solutions than using iframes. For example, if you have an Atoti+ license, you could use ActiveViam's UI SDK to:
- Add widgets connecting to atoti sessions in your own app.
- Extend the atoti web app with your own widgets / plugins (upcoming feature).

Could you tell us more about your use case so that we can help you better?

**Comment by donnalmh** [comment]
Hi Tibdex Thank you for your reply

We have the Atoti+ license already. I'm following the tutorial however how can we do this:
**Add widgets connecting to atoti sessions in your own app.**

The tutorial is based off its own source files. How do i integrate widgets from atoti sessions into our own app?

**Comment by tibdex** [comment]
The SDK is a [React](https://reactjs.org/) library so you can use React to mount an SDK `<Widget />` where you want in your existing app.

The `<Widget />`'s props are [documented in the API reference](https://activeviam.com/activeui/documentation/5.0.1/docs/api/components/#widget). You'll need some [Context providers](https://activeviam.com/activeui/documentation/5.0.1/docs/api/contextProviders) to be able to render a `<Widget />`.

You can still read the tutorial since it explains concepts such as [running a query](https://activeviam.com/activeui/documentation/5.0.1/docs/tutorial/runningAQuery) that you will need.

Tell me if this is not clear enough and we'll see if we can organize a training or something else to help you best.

**Comment by donnalmh** [comment]
Hi Thibault,

Thank you for the reply.

I saw the API docs and I also went through the tutorial. However, I feel the tutorial doesn't provide much introduction to the hierachy or structure of your overall API, but more of how to extend the Starter Kit UI. 

May I request a training session just to go through things such as adding in the <Dashboard> or the <Widget>? I tried to follow the storyboard and integrate them into my code as well but have met with error and also like you mention, I may not be aware that Context Providers are needed to wrap around certain components. 

If some form of training could be provided that would be helpful as the tutorial does not have much examples to integrate a Dashboard correctly from scratch.
